tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -463,28 +463,55 @@ mlir::Value MLIRScanner::createAllocOp(mlir::Type t, VarDecl *name,
 ValueCategory
 MLIRScanner::VisitExtVectorElementExpr(clang::ExtVectorElementExpr *expr) {
   auto base = Visit(expr->getBase());
+
   SmallVector<uint32_t, 4> indices;
   expr->getEncodedElementAccess(indices);
   assert(indices.size() == 1 &&
          "The support for higher dimensions to be implemented.");
-  auto loc = getMLIRLocation(expr->getExprLoc());
-  auto idx = castToIndex(getMLIRLocation(expr->getAccessorLoc()),
-                         builder.create<ConstantIntOp>(loc, indices[0], 32));
+
   assert(base.isReference);
   base.isReference = false;
-  auto mt = base.val.getType().cast<MemRefType>();
-  auto shape = std::vector<int64_t>(mt.getShape());
-  if (shape.size() == 1) {
-    shape[0] = -1;
-  } else {
-    shape.erase(shape.begin());
+
+  const auto et = base.val.getType();
+  assert(et.isa<LLVM::LLVMPointerType>() || et.isa<MemRefType>());
+
+  ValueCategory result = nullptr;
+  const auto exprLoc = getMLIRLocation(expr->getExprLoc());
+  const auto accLoc = getMLIRLocation(expr->getAccessorLoc());
+  const mlir::Value idxs[2] = {
```

Comment:
Can you move this inside the if pointer type to avoid generating these constants when not used?

Reply:
I can move the ```accLoc``` inside the future ```else if (memref)``` but the other constants are shared in both conditions.
Does it work for you?

Reply:
Oh you're right, I didn't see the use of idx in the memref branch. No need to move things then.